h/services/group.py

```diff
@@ -21,11 +22,42 @@ def __init__(self, session, user_fetcher):
         self.session = session
         self.user_fetcher = user_fetcher
 
-    def fetch(self, pubid):
-        """Fetch a group by ``pubid``"""
+    def fetch(self, pubid_or_groupid):
+        """
+        Fetch a group, using either a groupid or a pubid
+
+        :param pubid_or_groupid: a string in either :mod:`~h.pubid` format
+                                 or as :attribute:`~h.models.group.Group.groupid`
+        :rtype:`~h.models.group.Group` or None
+        """
 
+        if group_util.is_groupid(pubid_or_groupid):
+            return self.fetch_by_groupid(pubid_or_groupid)
+        return self.fetch_by_pubid(pubid_or_groupid)
+
+    def fetch_by_pubid(self, pubid):
+        """Return a group with the given ``pubid`` or None"""
         return self.session.query(Group).filter_by(pubid=pubid).one_or_none()
 
+    def fetch_by_groupid(self, groupid):
+        """
+        Return a group with the given ``groupid`` combination or None
+
+        :param groupid:     String in groupid format, e.g. ``group:foo@bar.com``
+                            See :mod:`~h.models.group.Group`
```

Comment:
Should be <code>:class:\`~h.models.Group\`</code> or <code>:attr:\`h.models.Group.groupid\`</code> (I'd omit the `~` in the second case).